homeassistant/components/intellifire/config_flow.py

```diff
@@ -4,60 +4,132 @@
+        """Handle local ip configuration."""
+
+        errors = {}
+        placeholder = {}
+        local_schema = vol.Schema({vol.Required(CONF_HOST): str})
+
+        if user_input is None:
+
+            current_hosts = [
+                entry.data[CONF_HOST]
+                for entry in self._async_current_entries(include_ignore=False)
+            ]
+
+            if self._discovered_hosts != []:
+                # Filter out already configured hosts
+                self._discovered_hosts = [
+                    ip for ip in self._discovered_hosts if ip not in current_hosts
+                ]
+
+                if len(self._discovered_hosts) > 1:
+                    return await self.async_step_pick_device()
+                if len(self._discovered_hosts) == 1:
+                    user_input = {CONF_HOST: self._discovered_hosts[0]}
+                    local_schema = vol.Schema(
+                        {vol.Required(CONF_HOST, default=user_input[CONF_HOST]): str}
+                    )
+                if len(self._discovered_hosts) == 0:
+                    # If there were discoverd hosts AND all got filtered out fire an error message
+                    errors["base"] = "already_discovered"
+                    LOGGER.debug("All discovered fireplaces have been configured")
+
+        else:  # There was User Input
+
+            placeholder = {CONF_HOST: user_input[CONF_HOST]}
+            try:
+                serial = await validate_host_input(user_input[CONF_HOST])
+            except (ConnectionError, ClientConnectionError):
+                errors["base"] = "cannot_connect"
+                local_schema = vol.Schema(
+                    {vol.Required(CONF_HOST, default=user_input[CONF_HOST]): str}
+                )
+            else:
+                await self.async_set_unique_id(serial)
+
+                # check if found before
+                self._abort_if_unique_id_configured(
+                    updates={
+                        CONF_HOST: user_input[CONF_HOST],
+                    }
+                )
+
+                return self.async_create_entry(
+                    title="Fireplace",
+                    data={
+                        CONF_HOST: user_input[CONF_HOST],
+                    },
+                )
+
+        return self.async_show_form(
+            step_id="local_config",
+            errors=errors,
+            description_placeholders=placeholder,
+            data_schema=local_schema,
+        )
+
+    async def async_step_pick_device(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
-        """Handle the initial step."""
-        if user_input is None:
-            return self.async_show_form(
-                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
-            )
-        errors = {}
+        """Pick which device to configure."""
+
+        if user_input is not None:
+            return await self.async_step_local_config(user_input=user_input)
```

Comment:
We shouldn't be passing the input between steps, instead move to the next step and get the input.

Reply:
```python
        if user_input is not None:
            # Update not_configured_hosts in order to trigger correct
            # behavior in next step
            self._not_configured_hosts = [user_input[CONF_HOST]]
            return await self.async_step_local_config()
```

Is this better?